pkg/provisioner/ironic/ironic.go

```diff
@@ -1151,6 +1151,42 @@ func (p *ironicProvisioner) GetFirmwareSettings(includeSchema bool) (settings me
 	return settings, schema, nil
 }
 
+// GetFirmwareComponentsForNode gets all available firmware components for a node and return a list.
+func (p *ironicProvisioner) GetFirmwareComponentsForNode() ([]metal3api.FirmwareComponentStatus, error) {
+	ironicNode, err := p.getNode()
+	if err != nil {
+		return nil, fmt.Errorf("could not get node to retrieve firmware components: %w", err)
+	}
+
+	// Get the components from Ironic via Gophercloud
+	componentList, componentListErr := nodes.ListFirmware(p.client, ironicNode.UUID).Extract()
```

Comment:
This is not going to have the right API version. You need to add it to AvailableFeatures and add a check here that the Ironic version even allows firmware.